[FIX] Allow to run jobs when Odoo is not listening on localhost

Garamotte opened the pull request:
On our dev boxes, we have Odoo running on loopback addresses other than localhost (ie. 127.16.4.2 for example), and `queue_job` had `localhost` hardcoded.

lmignon reviewed (approved):
@sylvain-garancher Thank you for the fix.
LGTM (Code review only)

simahawk reviewed (approved):
code LGTM

sbidoul commented:
@sylvain-garancher do you plan to update the documentation too?
Note there was extensive work on this topic in 9.0: https://github.com/OCA/connector/pull/247/files

guewen commented:
Yep I think the PR @sbidoul linked should be ported, a simple cherry-pick won't be enough though.

Garamotte commented:
Thanks, I didn't see the PR in 9.0 (because it was in another repo).
I'll update my PR when I have some time.

Commit:
[ADD] configurable scheme, host and http auth

Initial work by Nils Hamerlinck <nils@hamerlinck.fr> on the connector
module for Odoo v9.0 (commit 383959afc6f1f71a45320d9cab45b84a05572ac7)

Garamotte commented:
Git was smart enough to make most of the work with a cherry-pick, I mostly had to update the configuration part, but I didn't took time to test if everything works.

I credited the original author of this work on v9 in the commit message ;)
Thanks @sbidoul for the pointer.

guewen commented on queue_job/jobrunner/runner.py:
```diff
@@ -183,12 +195,15 @@ def set_job_pending():
     def urlopen():
-        url = ('http://localhost:%s/queue_job/runjob?db=%s&job_uuid=%s' %
-               (port, db_name, job_uuid))
+        url = ('%s://%s:%s/connector/runjob?db=%s&job_uuid=%s' %
```
s/connector/queue_job/

Garamotte replied:
Oops. Fixed :)

guewen reviewed (commented):
Great thanks! One remaining instance of 'connector' slipped through

Garamotte commented:
@sbidoul I think this PR can be merged now.

aliencrash commented:
How we have to configure to use this feature?
